torch/fx/experimental/proxy_tensor.py

```diff
@@ -410,3 +410,47 @@ def wrap_fake(x):
         return t
 
     return wrapped
+
+
+def _find_proxy_tensor(*objects_to_search):
+    # return first proxy tensor found or None
+    proxy_tensors = (o for o in objects_to_search if isinstance(o, ProxyTensor))
+    return next(proxy_tensors, None)
+
+
+def get_isolated_graphmodule(func, args, kwargs):
+    """A helper function used to get the GraphModule for the given func.
+
+    It's expected to be used in the ProxyTensor tracing context.
+    It detaches the args and kwargs from the current tracer so that the trace of
+    the current graph module can be created without any side-effects.
+    """
+    # make_fx doesn't support kwargs, so we need to do this flattening
+    # and then unflatten the args before calling func
+    all_args, spec = pytree.tree_flatten((args, kwargs))
+
+    def wrapped(args):
+        fn_args, fn_kwargs = pytree.tree_unflatten(args, spec)
+        return func(*fn_args, **fn_kwargs)
+
+    # extract old tracer object
+    proxy_tensor = _find_proxy_tensor(*all_args)
+    if proxy_tensor is not None:
+        old_tracer = proxy_tensor.proxy.tracer
+
+    # create a new tracer object
+    graph = torch.fx.Graph()
+    new_tracer = PythonKeyTracer()
+    new_tracer.graph = graph
+
+    try:
+        for arg in all_args:
+            if isinstance(arg, ProxyTensor):
+                arg.proxy.tracer = new_tracer
```

Comment:
Instead of mutating the argument, couldn't you pull out the inner `elem` and create a fresh ProxyTensor from that? Then you wouldn't need to worry about resetting the proxy afterwards.

Reply:
Yes I can and I posted this example of creating a fresh ProxyTensor on Slack (https://pytorch.slack.com/archives/C03DP57R27M/p1658256451021309?thread_ts=1658256439.210389&cid=C03DP57R27M)

It requires creating a fresh torch.fx.proxy.Proxy which in turn requires creating a fresh Node (because nodes do not work with arbitrary graphs, it has to match). Just resetting the tracer seemed cleaner. Would you like me to change that and create a fresh ProxyTensor instead?

Reply:
I changed it now to create a fresh ProxyTensor instead of mutating.